### PR TITLE
feat(file-upload): identify and treat the H5P files in the general file upload

### LINKED ISSUE #1368: upload a h5p should create a h5p type item
Using /upload with a h5p does not create a correct h5p file, but a s3File.
From the extension I believe it's possible to redirect the upload to "import h5p"

Question: should we convert incorrect s3File h5p to "h5p" files?

### PR DESCRIPTION
**Pull request by MatveyK**
Closes #1368

### CODE REVIEW

**Review by pyphilia — APPROVED** [review]
LGTM! 🎨

File: src/services/item/plugins/file/index.ts [hunk 1]
```diff
@@ -134,14 +137,30 @@ const basePlugin: FastifyPluginAsyncTypebox<GraaspPluginFileOptions> = async (fa
...
           try {
-            const i = await fileItemService.upload(member, repositories, {
-              parentId,
-              filename,
-              mimetype,
-              stream,
-              previousItemId,
-            });
-            items.push(i);
+            // if the file is an H5P file, we treat it apporpriately
```
[the remark this answers is not in the document]
Comment (MatveyK) [reply src/services/item/plugins/file/index.ts:140]: Good catch.

File: src/services/item/plugins/file/test/index.test.ts [hunk 2]
```diff
@@ -253,6 +253,37 @@ describe('File Item routes tests', () => {
...
+          const response = await app.inject({
+            method: HttpMethod.Post,
+            url: `${ITEMS_ROUTE_PREFIX}/upload`,
+            payload: form,
+            headers: form.getHeaders(),
+          });
+
+          // check the response value
+          const newItems = Object.values(response.json().data) as Item[];
+          expect(response.statusCode).toBe(StatusCodes.OK);
```
[the remark this answers is not in the document]
Comment (MatveyK) [reply src/services/item/plugins/file/test/index.test.ts:272]: Indeed.

File: src/services/item/plugins/file/test/index.test.ts [hunk 3]
```diff
@@ -253,6 +253,37 @@ describe('File Item routes tests', () => {
...
+          expect(newItems.length).toBe(2);
+
+          // check that both items exist in db and that their types are correctly interpreted
+          const imageItem = await testUtils.rawItemRepository.findOneBy({ id: newItems[0].id });
+          expectItem(imageItem, newItems[0]);
+          expect(
+            imageItem?.type === ItemType.LOCAL_FILE || imageItem?.type === ItemType.S3_FILE,
```
[the remark this answers is not in the document]
Comment (MatveyK) [reply src/services/item/plugins/file/test/index.test.ts:279]: I have put in a check that verifies if the `S3_FILE_ITEM_PLUGIN` is true or not. Not sure if there's a more elegant way of doing this.

**Review by spaenleh — APPROVED** [review]
👍

**Pull request merged**